src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java

```diff
@@ -47,6 +49,11 @@ class StreamingIngestStage {
   private static final long REFRESH_THRESHOLD_IN_MS =
       TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
 
+  // Stage credential refresh interval, currently the token will expire in 1hr for GCS and 2hr for
```

Comment:
Should we pass back the credential duration in the client configure response instead of hardcoding it here? That would be more future-proof.

Reply:
This seems to be an overkill for me, because
- This is just to fix a complain from customer that they don't like to see the ERROR log in their log file. Even if the frequency gets smaller in the future, everything will continue to work, except customers will see the log again.
- The parameter not user configurable, and the last time it gets updated was back in 2021
- Even if we support this, they will get the new parameter values only if they restart the client

In general, I agree it's better to do what you suggested. We have discussed in the past to support something like this for all other parameters so that we can control other behaviors of the SDK without a client side upgrade/downgrade, but we decided to NOT do it and focus on ROWSET API. WDYT?